mailpoet/assets/js/src/automation/editor/components/actions/trash-button.tsx

```diff
@@ -0,0 +1,70 @@
+import apiFetch from '@wordpress/api-fetch';
+import { Button } from '@wordpress/components';
+import { StoreDescriptor, useDispatch, useSelect } from '@wordpress/data';
+import { store as noticesStore } from '@wordpress/notices';
+import { addQueryArgs } from '@wordpress/url';
+import { confirmAlert } from '../../../../common/confirm_alert';
+import { store } from '../../store';
+import { Workflow } from '../workflow/types';
+import { MailPoet } from '../../../../mailpoet';
+import { LISTING_NOTICE_PARAMETERS } from '../../../listing/workflow-listing-notices';
+
+export function TrashButton(): JSX.Element {
+  const { createErrorNotice } = useDispatch(noticesStore as StoreDescriptor);
+  const { workflow } = useSelect(
+    (select) => ({
+      workflow: select(store).getWorkflowData(),
+    }),
+    [],
+  );
+
+  const trash = () => {
+    apiFetch({
+      path: `/workflows/${workflow.id}`,
+      method: 'PUT',
+      data: {
+        ...workflow,
+        status: 'trash',
+      },
+    })
+      .then(({ data }: { data: Workflow }) => {
+        if (data.status !== 'trash') {
+          void createErrorNotice('An error occurred!', {
+            explicitDismiss: true,
+          });
+          return;
+        }
+
+        window.location.href = addQueryArgs(MailPoet.urls.automationListing, {
+          [LISTING_NOTICE_PARAMETERS.workflowDeleted]: workflow.id,
+        });
+      })
+      .catch((): void => {
+        void createErrorNotice('An error occurred!', {
+          explicitDismiss: true,
+        });
+      });
+
+    return true;
+  };
```

Comment:
@websupporter Another thing I noticed is that all of this should be an action in the editor store. Just to let you know, fixing in my branch.